proxy.go

```diff
@@ -1,96 +1,111 @@
+func ProxyHandler(w http.ResponseWriter, req *http.Request) {
+	Logger.Println("New request from:", req.RemoteAddr, "for URI:", req.URL.String())
 
-		// Retry with server credentials if authentication is required
-		if r.StatusCode == 401 {
-			// for debugging
-			defer TimeTrack(time.Now(), "Fetching")
-			var resp *http.Response
-			var client *http.Client
-			var solutionMsg string
-			req, err := http.NewRequest("GET", ctx.Req.URL.String(), ctx.Req.Body)
-			req.Header.Add("On-Behalf-Of", conf.User)
-			// Retry the request
-			if len(cookies[conf.User]) > 0 { // Use existing cookie
-				req.AddCookie(cookies[conf.User][ctx.Req.Host][0])
-				// Create the client
-				client = &http.Client{
-					Transport: &http.Transport{
-						TLSClientConfig: &tls.Config{
-							InsecureSkipVerify: conf.InsecureSkipVerify,
-						},
-					},
-				}
-				solutionMsg = "Retrying with cookies"
-			} else { // Using WebIDTLS client
-				client = webidTlsClient
-				solutionMsg = "Retrying with WebID-TLS"
-			}
-			resp, err = client.Do(req)
-			if err != nil {
-				Logger.Fatal("GET:", err)
-			}
-			// Store cookies per user and request host
-			if len(resp.Cookies()) > 0 {
-				cookiesL.Lock()
-				// Should store cookies based on domain value AND path from cookie
-				cookies[conf.User] = map[string][]*http.Cookie{}
-				cookies[conf.User][ctx.Req.Host] = resp.Cookies()
-				Logger.Printf("Cookies: %+v\n", cookies)
-				cookiesL.Unlock()
-			}
-			Logger.Println("Resource "+ctx.Req.URL.String(),
-				"requires authentication (HTTP 401).", solutionMsg,
-				"resulted in HTTP", resp.StatusCode)
-			return resp
-		}
-		Logger.Println("Received data with status HTTP", r.StatusCode)
-		return r
-	})
-}
+	user := req.Header.Get("User")
+	// override if we have specified a user in config
+	if len(userWebID) > 0 {
+		user = userWebID
+	}
 
-func ProxyHandler(w http.ResponseWriter, req *http.Request) {
-	toProxy := req.FormValue("uri")
-	if len(toProxy) == 0 {
+	uri := req.FormValue("uri")
+	if len(uri) == 0 {
+		msg := "No URI was provided to the proxy!"
+		Logger.Println(msg, req.URL.String())
 		w.WriteHeader(500)
-		w.Write([]byte("No URI was provided to the proxy!"))
+		w.Write([]byte(msg))
 		return
 	}
 
-	uri, err := url.ParseRequestURI(toProxy)
+	resource, err := url.ParseRequestURI(uri)
 	if err != nil {
 		w.WriteHeader(500)
 		w.Write([]byte("Error parsing URL: " + req.URL.String() + " " + err.Error()))
 		Logger.Println("Error parsing URL:", req.URL, err.Error())
 		return
 	}
-	req.URL = uri
-	req.Host = uri.Host
-	req.RequestURI = uri.RequestURI()
-	Logger.Println("Proxying request for:", req.URL)
-	proxy.ServeHTTP(w, req)
+	req.URL = resource
+	req.Host = resource.Host
+	req.RequestURI = resource.RequestURI()
+	Logger.Println("Proxying request for URI:", req.URL, "and user:", user)
+
+	// build new response
+	plain, err := http.NewRequest("GET", req.URL.String(), req.Body)
+	if err != nil {
+		Logger.Fatal("GET error:", err)
```

Comment:
In fact, that error will never occur in this case. I'm considering skipping the check altogether.